health/collector_test.go

```diff
@@ -0,0 +1,127 @@
+package health_test
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/zenoss/zenoss-go-sdk/health"
+	"github.com/zenoss/zenoss-go-sdk/health/target"
+	"github.com/zenoss/zenoss-go-sdk/health/writer"
+	"github.com/zenoss/zenoss-go-sdk/health/writer/mocks"
+)
+
+var _ = Describe("Collector", func() {
+
+	var (
+		err       error
+		ctx       context.Context
+		cancel    context.CancelFunc
+		config    *health.Config
+		targets   []*target.Target
+		dest      *mocks.Destination
+		wr        writer.HealthWriter
+		manager   health.Manager
+		collector health.Collector
+		msg       *target.Message
+	)
+
+	const (
+		testTargetID = "test.target"
+
+		testMetricID   = "test.metric"
+		testCounterID  = "test.counter"
+		totalCounterID = "total.counter"
+	)
+
+	BeforeEach(func() {
+		config = health.NewConfig()
+		config.CollectionCycle = 1 * time.Second
+
+		tar, _ := target.New(
```

Comment:
This test doesn't check whether the target is valid and whether it was added to manager. ie no need to add target to the manager if we don't check it during the test

(Anyway you added a target with ID "1" when you registering metrics under "test.target" ID